code/modules/awaymissions/mission_code/runner.dm

```diff
@@ -0,0 +1,134 @@
+
+/obj/effect/portal/permanent/one_way/recall/pit_faller/teleport(atom/movable/M, force = FALSE)
+	if(GLOB.vr_runner_active)
+		return FALSE
+	if(!ishuman(M))
+		return FALSE
+	. = ..()
+	if(.)
+		var/mob/living/carbon/human/H = M
+		GLOB.vr_runner_players += H
+		if(!game_starting)
+			INVOKE_ASYNC(src, .proc/game_start_countdown)
+
+/obj/effect/portal/permanent/one_way/recall/pit_faller/recall_effect(mob/user)
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		GLOB.vr_runner_players -= H
+		if(GLOB.vr_runner_active)
+			if(GLOB.vr_runner_players.len == 0)
+				to_chat(H, "<span class='notice'>You win! You survived for [(world.time - game_start_time) / 10] seconds.</span>")
+				end_game()
+			else
+				to_chat(H, "<span class='notice'>You survived for [(world.time - game_start_time) / 10] seconds. [GLOB.vr_runner_players.len] other player(s) remained.</span>")
+
+/obj/effect/portal/permanent/one_way/recall/pit_faller/proc/game_start_countdown(wait_seconds = 10)
+	game_starting = 1
+	for(var/seconds_remaining = wait_seconds to 1 step -1)
+		if(GLOB.vr_runner_players.len == 0)
+			game_starting = 0
+			return FALSE
+		for(var/mob/living/carbon/human/H in GLOB.vr_runner_players)
+			to_chat(H, "<span class='notice'>Game starting in [seconds_remaining].</span>")
+		sleep(10)
+	if(GLOB.vr_runner_players.len > 0)
+		GLOB.vr_runner_active = 1
+		for(var/mob/living/carbon/human/H in GLOB.vr_runner_players)
+			to_chat(H, "<span class='notice'>Game Started!</span>")
+			var/turf/open/indestructible/runner/R = get_turf(H)
+			INVOKE_ASYNC(R, /turf/open/indestructible/runner.proc/turf_fall)
+		color = COLOR_RED
+		INVOKE_ASYNC(src, .proc/random_falling_tiles)
+	game_starting = 0
+	game_start_time = world.time
+
+/obj/effect/portal/permanent/one_way/recall/pit_faller/proc/random_falling_tiles()
+	if(!GLOB.vr_runner_active)
+		return FALSE
+	for(var/turf/open/indestructible/runner/R in GLOB.vr_runner_tiles)
+		if(R.color != COLOR_ALMOST_BLACK && prob(1))
+			INVOKE_ASYNC(R, /turf/open/indestructible/runner.proc/turf_fall)
+	sleep(fall_wait)
+	random_falling_tiles()
+
+/obj/effect/portal/permanent/one_way/recall/pit_faller/proc/end_game()
+	color = initial(color)
+	GLOB.vr_runner_active = 0
+	game_starting = 0
+	for(var/turf/open/indestructible/runner/R in GLOB.vr_runner_tiles)
+		R.reset_fall()
+
+/obj/effect/portal/permanent/one_way/destroy/pit_faller
+	name = "Runner Exit Portal"
+	id = "vr runner"
+
+/turf/open/indestructible/runner
+	name = "Shaky Ground"
+	desc = "If you walk on that that you better keep running!"
+	var/not_reset = 0 // check so turfs dont fall after being reset
+	var/falling_time = 15 // time it takes for the turf to fall
+
+/turf/open/indestructible/runner/Initialize()
+	. = ..()
+	GLOB.vr_runner_tiles += src
+
+/turf/open/indestructible/runner/Entered(atom/movable/A)
+	. = ..()
+	if(isliving(A) && GLOB.vr_runner_active)
+		if(color == COLOR_ALMOST_BLACK)
```

Comment:
> It's a color check

Reply:
doesn't matter since there's no way to get any color changing stuff in there because it equips and gets rid of your equipment suck my ass